### PR TITLE
Remove unnecessary NOTREACHED_IN_MIGRATION

### PR DESCRIPTION
**Pull request by mherrmann**
Necessary cleanup following some upstream changes. Cf. https://bravesoftware.slack.com/archives/C7VLGSR55/p1730902320819289?thread_ts=1716906901.716109&cid=C7VLGSR55.

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

It's not necessary to test this separately, as long as updates are being checked on macOS.

### CODE REVIEW

File: chromium_src/chrome/browser/ui/webui/help/version_updater_mac.mm [hunk 1]
```diff
@@ -171,7 +171,7 @@ - (void)handleStatusNotification:(NSNotification*)notification {
       break;
 
     default:
-      NOTREACHED_IN_MIGRATION();
+      // We are not currently interested in this status.
```
Comment (iefremov) [inline chromium_src/chrome/browser/ui/webui/help/version_updater_mac.mm:174]: actually it's not really clear what status is assumed and why we are not interested.

Comment (bsclifton) [reply chromium_src/chrome/browser/ui/webui/help/version_updater_mac.mm:174]: I guess it's the same value as what happens above for `kAutoupdateAvailable`. But yeah - it's not clear what the default value for `Status` enum is. Would probably be good to VLOG out a warning and what value was received. Something like:
```
VLOG(2) << "Unexpected value sparkle_status=" << sparkle_status << ". Ignoring."
```

Comment (simonhong) [reply chromium_src/chrome/browser/ui/webui/help/version_updater_mac.mm:174]: In our code, we don't use other status.
This status is determined and notified by `SparkleGlue` and `SparkeGlue` only uses above listed status.
So, it's not possible to reach to `default` case.

When I ported to sparkle at first, our status enum was below and all status was handled above.
```
enum AutoupdateStatus {
  kAutoupdateNone = 0,        // no version (initial state only)
  kAutoupdateRegistering,     // no version (asynchronous operation in progress)
  kAutoupdateRegistered,      // no version
  kAutoupdateChecking,        // no version (asynchronous operation in progress)
  kAutoupdateCurrent,         // version of the running application
  kAutoupdateAvailable,       // version of the update that is available
  kAutoupdateInstalling,      // no version (asynchronous operation in progress)
  kAutoupdateInstalled,       // version of the update that was installed
  kAutoupdateRegisterFailed,  // no version
  kAutoupdateCheckFailed,     // no version
  kAutoupdateInstallFailed,   // no version
};
```
and later, we refactored and we shared upstream's keystone glue's AutoupdateStatus.
It's below.
```
enum AutoupdateStatus {
  kAutoupdateNone = 0,        // no version (initial state only)
  kAutoupdateRegistering,     // no version (asynchronous operation in progress)
  kAutoupdateRegistered,      // no version
  kAutoupdateChecking,        // no version (asynchronous operation in progress)
  kAutoupdateCurrent,         // version of the running application
  kAutoupdateAvailable,       // version of the update that is available
  kAutoupdateInstalling,      // no version (asynchronous operation in progress)
  kAutoupdateInstalled,       // version of the update that was installed
  kAutoupdatePromoting,       // no version (asynchronous operation in progress)
  kAutoupdatePromoted,        // no version
  kAutoupdateRegisterFailed,  // no version
  kAutoupdateCheckFailed,     // no version
  kAutoupdateInstallFailed,   // no version
  kAutoupdatePromoteFailed,   // no version
  kAutoupdateNeedsPromotion,  // no version
};
```

Although, keystone glue's AutoupdateStatus includes some other state(`kXXXPromoXX`) that are not handled in our switch statement, we can ignore it as we don't use that status from our `SparkleGlue`.

So, I recommend to update here 
from 
```
      NOTREACHED_IN_MIGRATION();
      // We are not currently interested in this status.
```
to
```
     // We don't use other status.
     NOTREACHED();
```

**Review by simonhong — APPROVED** [review]
++

**Review by cdesouza-chromium — APPROVED** [review]
LGTM

**Pull request merged**